programs/account-compression/src/instructions/append_leaves.rs

```diff
@@ -74,18 +76,31 @@ pub fn process_append_leaves_to_merkle_trees<'a, 'b, 'c: 'info, 'info>(
     #[cfg(target_os = "solana")]
     light_heap::GLOBAL_ALLOCATOR.log_total_heap("append_leaves: merkle tree map");
 
-    let mut changelog_events = Vec::with_capacity(leaves.len());
-    for (mt, leaves) in merkle_tree_map.values() {
+    let mut leaves_start = 0;
+    let mut changelog_events = Vec::with_capacity(BATCH_SIZE);
+
+    let mut merkle_tree_map_iter = merkle_tree_map.values();
+    let mut mt_next = merkle_tree_map_iter.next();
+
+    // for (mt, leaves) in merkle_tree_map.values() {
+    while let Some((mt, leaves)) = mt_next {
         let merkle_tree_account = AccountLoader::<StateMerkleTreeAccount>::try_from(mt).unwrap();
         let merkle_tree_pubkey = merkle_tree_account.key();
         let mut merkle_tree = merkle_tree_account.load_mut()?;
 
         check_registered_or_signer::<AppendLeaves, StateMerkleTreeAccount>(&ctx, &merkle_tree)?;
 
         // msg!("inserting leaves: {:?}", leaves);
+
+        let leaves_end = if leaves.len() < BATCH_SIZE {
+            leaves.len() - leaves_start
```

Comment:
what if ``leaves.len() < leaves_start`` ?
I think this case can occur, for example:
MT 1: has 7 leaves
MT 2: has 1 leaf

Reply:
you could do:
``
// if leaves len is less than remaining batch size, return leaves len
// if leaves len is more than remaining batch size return remaining batch size
if leaves.len() < BATCH_SIZE - leaves_start {
  leaves.len()
  } else {
      BATCH_SIZE - leaves_start
  };

Reply:
Ok, good point